new_dawn_server/users/api/resources.py

```diff
@@ -279,6 +281,15 @@ class Meta:
         queryset = Profile.objects.all()
         resource_name = "profile"
 
+    def apply_filters(self, request, applicable_filters):
+        ranking = request.GET.get('ranking', False)
+        viewer_id = request.GET.get('viewer_id','')
+        filtered = super(ProfileResource, self).apply_filters(request, applicable_filters)
+        if viewer_id:
```

Comment:
You will be able to pass the test if you put if view_id logic inside if ranking.